### PR TITLE
Backup: Custom storage volumes export & import

### PR DESCRIPTION
**Pull request by tomponline**
Replacement for https://github.com/lxc/lxd/pull/7837

- Adds `TypeCustom` to indicate custom volume backup in `backup.Info.Type` field.
- Adds `Config` field to `backup.Info` struct in order to store the equivalent of backup.yaml inside index.yaml for custom volumes.
- Adds `VolumeSnapshots` field to `backup.Config` struct in order to store custom volume snapshot config in backup.
- Stores backups in `${LXD_DIR}/backups/custom/<pool>/<project>_<volume>/<backupName>`
- Adds `lxc storage volume export` and `lxc storage volume import` commands.
- Deleting a custom volume will delete its backups (like instances).
- Renaming a custom volume will rename its backups (both location on disk and name prefix in DB).
- Volume config and snapshot config is restored (however incompatible entries when importing into a different type of storage pool are removed).

### CODE REVIEW

**Comment by tomponline** [comment]
@stgraber I ended up git resetting the previous PR entirely and then recommitting everything as new commits to allow me to break things down into smaller chunks. However where I've not made any changes to a code block (or made only a small change) I added a `Co-authored-by` for @monstermunchkin  to the git commit message to maintain attribution.

**Comment by stgraber** [comment]
Probably can be rebased now?

**Comment by tomponline** [comment]
@stgraber rebased

File: lxc/storage_volume.go [hunk 1]
```diff
@@ -1631,3 +1643,228 @@ func (c *cmdStorageVolumeRestore) Run(cmd *cobra.Command, args []string) error {
...
+	storageVolume *cmdStorageVolume
+
+	flagVolumeOnly           bool
+	flagOptimizedStorage     bool
+	flagCompressionAlgorithm string
+}
+
+func (c *cmdStorageVolumeExport) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = i18n.G("export [<remote>:]<pool> <volume> [<path>]")
```
Comment (stgraber) [inline lxc/storage_volume.go:1660]: Should use `usage()`

File: lxc/storage_volume.go [hunk 2]
```diff
@@ -1631,3 +1643,228 @@ func (c *cmdStorageVolumeRestore) Run(cmd *cobra.Command, args []string) error {
...
+// Import
+type cmdStorageVolumeImport struct {
+	global        *cmdGlobal
+	storage       *cmdStorage
+	storageVolume *cmdStorageVolume
+}
+
+func (c *cmdStorageVolumeImport) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = i18n.G("import [<remote>:]<pool> <backup file>")
```
Comment (stgraber) [inline lxc/storage_volume.go:1795]: should use `usage()`

**Pull request merged**